libbeat/docs/command-reference.asciidoc

```diff
@@ -0,0 +1,583 @@
+{beatname_lc} run --modules nginx,mysql,system
+-----
++
+Rather than specifying the list of modules every time you run {beatname_uc},
+you can use the <<modules-command,`modules`>> command to enable and disable
+specific modules. Then when you run {beatname_uc}, it will run any modules
+that are enabled.
+
+//REVIEWERS: Is the above syntax correct for --modules? We used to say -modules=nginx,mysql,system. Not sure if it makes a difference, but I'm trying to be consistent with what users see when they view --help.
+
+endif::[]
+
+ifeval::["{beatname_lc}"=="filebeat"]
+
+*`--once`*::
+When the `--once` flag is used, {beatname_uc} starts all configured harvesters
+and prospectors, and runs each prospector until the harvesters are closed. If
+you set the `--once` flag, you should also set `close_eof` so the harvester is
+closed when the end of the file is reached. By default harvesters are closed
+after `close_inactive` is reached.
+
+
+//REVIEWERS: Is the behavior of the harvesters/prospectors described above still true?
+
+endif::[]
+
+//REVIEWERS: I'm not documenting --setup because it will be removed in 6.0. Are there any other flags that I need to remove? Or mention that they're being maintained for backwards compability?
+
+ifeval::["{beatname_lc}"=="metricbeat"]
+
+*`--system.hostfs MOUNT_POINT`*::
+
+Specifies the mount point of the host's filesystem for use in monitoring a host
+from within a container.
+
+endif::[]
+
+ifeval::["{beatname_lc}"=="packetbeat"]
+
+*`-t`*::
+Reads packets from the pcap file as fast as possible without sleeping. Use this
+option in combination with the `-I` option. The `-t` option is useful only for
+testing Packetbeat.
+
+endif::[]
+
+{global-flags}
+
+*EXAMPLE*
+
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} run -e
+-----
+
+
+[[setup-command]]
+==== `setup` command
+
+{setup-command-short-desc}.
+
+* The index template ensures that fields are mapped correctly in Elasticsearch.
+* The Kibana dashboards make it easier for you to visualize {beatname_uc} data
+in Kibana.
+* The machine learning jobs contain the configuration information and metadata
+necessary to analyze data for anomalies.
+
+//REVIEWERS: What are the requirements/permissions for users who want to load machine learning jobs? We need to document that here.
+
+*SYNOPSIS*
+
+["source","sh",subs="attributes"]
+----
+{beatname_lc} setup [FLAGS]
+----
+
+
+*FLAGS*
+
+*`--dashboards`*::
+Sets up the Kibana dashboards only.
+
+*`-h, --help`*::
+Shows help for the `setup` command.
+
+*`--machine-learning`*::
+Sets up machine learning job configurations only.
+
+ifeval::["{beatname_lc}"=="filebeat"]
+
+*`--modules MODULE_LIST`*::
+Sets up the environment (index template, dashboards, etc) for the modules
+specified in the comma-separated list.
+
+//REVIEWERS: Is the above description correct? The help is unclear on what this sets up. Makes it sound like the command enables the modules. (Ran out of time to test this on my own...my apologies.)
```

Comment:
Yep, setup configures all dashboards, and I think it's the same for ML jobs. We only allow this flag to avoid errors in case no module is defined in filebeat.yml. To be consistent with `run` command. But it doesn't make a difference on what is configured.